docs/how_to/namespaced_apphooks.rst

```diff
@@ -1,45 +1,148 @@
+* It sets the *default apphook instance name* (which you'll see in the *Application instance name* field).
+
+We'll explain these with an example. Let's suppose that your application's views or templates use
+``reverse('myapp:index')`` or ``{% url 'myapp:index' %}``.
+
+In this case the namespace of any apphooks you apply must match ``myapp``. If they don't, your pages using them will
+throw up a ``NoReverseMatch`` error.
+
+You can set the namespace for the instance of the apphook in the *Application instance name* field. However, you'll
+need to set that to something *different* if an instance with that value already exists. In this case, as long as
+``app_name = "myapp"`` it doesn't matter; even if the system doesn't find a match with the name of the instance it will
+fall back to the one hard-wired into the class.
+
+In other words setting ``app_name`` correctly guarantees that URL-reversing will work, because it sets the fallback
+namespace approppriately.
+
+
+Set a namespace at instance-level
+=================================
+
+On the other hand, the *Application instance name* will override the ``app_name`` *if* a match is found.
+
+This arrangement allows you to use multiple application instances and namespaces if that flexibility is required, while
+guaranteeing a simple way to make it work when it's not.
+
+Django's :ref:`django:topics-http-reversing-url-namespaces` documentation provides more information on how this works,
+but the simplified version is:
+
+1. First, it'll try to find a match for the *Application instance name*.
+2. If it fails, it will try to find a match for the ``app_name``.
+
+
+.. _apphook_configurations:
+
+**********************
+Apphook configurations
+**********************
+
+Namespacing your apphooks also makes it possible to manage additional database-stored aphook configuration, on an
+instance-by-instance basis.
 
-We'll illustrate this example with a new application.
 
 Basic concepts
-##############
+==============
+
+To capture the configuration that different instances of an apphook can take, a Django model needs to be created - each
+apphook instance will be an instance of that model, and administered through the Django admin in the usual way.
+
+Once set up, an apphook configuration can be applied to to an apphook instance, in the *Advanced settings* of the page
+the apphook instance belongs to:
+
+.. image:: /how_to/images/select_apphook_configuration.png
+   :alt: selecting an apphook configuration application
+   :width: 400
+   :align: center
+
+The configuration is then loaded in the application's views for that namespace, and will be used to determined how it
+behaves.
+
+Creating an application configuration in fact creates an apphook instance namespace. Once created, the namespace of a
+configuration cannot be changed - if a different namespace is required, a new configuration will also need to be
```

Comment:
"cannot" is a strong word here. "shouldn't" is probably more apt. However, Aldryn Apphooks Config does make it read-only.